Sources/SwiftFormat/Utilities/URL+isRoot.swift

```diff
@@ -12,21 +12,43 @@
 
 import Foundation
 
+#if os(Windows)
+import WinSDK
+#endif
+
 extension URL {
+  /// Returns a `Bool` to indicate if the given `URL` leads to the root of a filesystem.
+  /// A non-filesystem type `URL` will always return false.
   @_spi(Testing) public var isRoot: Bool {
+    guard isFileURL else { return false }
+
+    #if os(macOS)
+    return self.path == NSOpenStepRootDirectory()
+    #endif
+
+    #if compiler(>=6.1)
```

Comment:
I would change this to `#elseif compiler(>=6.1)` to ensure we don’t get any of the returns below for macOS